platform/saivpp/vpplib/SwitchStateBaseRif.cpp

```diff
@@ -615,6 +755,157 @@ sai_status_t SwitchStateBase::vpp_add_del_intf_ip_addr_norif (
+        int ret = create_loopback_instance(hw_ifname, instance);
+        if (ret != 0) {
+            SWSS_LOG_ERROR("create_loopback_instance returned error");
+        }
+ 
+        /* Get new list of physical interfaces from VPP */
+        refresh_interfaces_list();
+        
+        vpp_ip_route_t ip_route;
+        create_route_prefix(&route_entry, &ip_route);
+
+        char prefixIp4Str[INET_ADDRSTRLEN];
+        inet_ntop(AF_INET, &(ip_route.prefix_addr.addr.ip4.sin_addr), prefixIp4Str, INET_ADDRSTRLEN);
+
+        SWSS_LOG_DEBUG("hw_ifname:%s is_add:%u", hw_ifname, is_add);
+        ret = interface_ip_address_add_del(hw_ifname, &ip_route, is_add);
+        if (ret != 0) {
+            SWSS_LOG_ERROR("interface_ip_address_add_del returned error");
+        }
+
+        //Set state up
+        bool is_up = true;
+        interface_set_state(hw_ifname, is_up);
+
+        const std::string hostIfname = findInterfaceNameByIp(destinationIP);
+        lpbIpToHostIfMap[destinationIP] = hostIfname;
+        
+        // remove host looback interface before creating lcp tap
+        bool lpb_add = false;
+        const std::string destinationIp = destinationIP;
+        int result = configureLoopbackInterface(lpb_add, hostIfname, destinationIp, ip_route.prefix_len);
+        if (result != 0) 
+        {
+            SWSS_LOG_ERROR("Failed to configure loopback interface remove");
+        }
+
+        // create lcp tap between vpp and host
+        {
+            const char *sonic_name = hostIfname.c_str();
+            const char *vpp_name = interfaceName.c_str();
+	        init_vpp_client();
+            SWSS_LOG_DEBUG("configure_lcp_interface vpp_name:%s sonic_name:%s", vpp_name, sonic_name);
+	        configure_lcp_interface(vpp_name, sonic_name, is_add);
+        }
+
+        // add back host looback interface after creating lcp tap
+        lpb_add = true;
+        result = configureLoopbackInterface(lpb_add, hostIfname, destinationIp, ip_route.prefix_len);
+        if (result != 0) 
+        {
+            SWSS_LOG_ERROR("Failed to configure loopback interface add");
+        }
+
+        lpbIpAddProgressMap[destinationIP] = false;
+        return SAI_STATUS_SUCCESS;
+
+    } else 
+    {
+        // Delete loopback interface
+        if(lpbIpAddProgressMap[destinationIP] == true)
+        {
+            return SAI_STATUS_SUCCESS;
+        }
+        if(lpbIpDelProgressMap[destinationIP] == true)
+        {
+            return SAI_STATUS_SUCCESS;
+        }
+        if (lpbIpExistMap[destinationIP] != true)
+        {
+            return SAI_STATUS_SUCCESS;
+        }
+
+        lpbIpDelProgressMap[destinationIP] = true;
+        std::string interfaceName = lpbIpToIfMap[destinationIP];
+        const std::string hostIfname = lpbIpToHostIfMap[destinationIP];
+        uint32_t instance = lpbInstMap[interfaceName];
+        const char *hw_ifname = interfaceName.c_str();
+
+        // remove lcp tap between vpp and host
+        {
```

Comment:
deleting the loopback should also delete lcp pair by the VPP. You may not need to explicitly call delete lcp here.

Reply:
Fixed.